src/components/Table/utils/TableState.ts

```diff
@@ -46,21 +47,30 @@ export class TableState {
   private initialSortState: SortState | undefined;
   private onSort: ((orderBy: any | undefined, direction: Direction | undefined) => void) | undefined;
 
+  // Non-reactive list of our columns
+  public columns: GridColumnWithId<any>[] = [];
+  // An observable set of column ids to keep track of which columns are currently expanded
+  private expandedColumns = new ObservableSet<string>();
+  // An observable set of column ids to keep track of which columns are visible
+  public visibleColumns = new ObservableSet<string>();
+  private visibleColumnStorageKey: string = "";
```

Comment:
Question: I don't see any `if key === "" don't store`; is the intent to always persist visible columns into this `""` spot?

Fwiw I probably would not have minded skipping the storage key stuff in v1, if anything b/c my suspicion/bias is that we introduced these sort of "remember the page is sorted/filtered/etc by x" for the original S&S view that was very "hop to new page to see details, hit back, hop to a new page, hit back" happy; but given that most of our views have superdrawers to enable working on the page/without leaving the page, I just wonder if the "remember last-used sort/filter/collapse state" is not as important as it used to be.

Reply:
This session storage for visible columns is an existing behavior that the `useColumns` hook was doing for us. But now we don't have the `useColumns` hook and the `EditColumnsButton` utilizes the API.

Reply:
The storage key also defaults to a value based on the Table's column ids. It is initialized in `setColumns`